### ISSUE TITLE
Use the hoogle library in the hoogle binary

### ISSUE THREAD

**Issue by nomeata** [ndmitchell/hoogle#118]
Hi,

while uploading a new version of hoogle to Debian I noticed that it re-builds the modules when building the executable. Wouldn’t it make sense to have the executable depend on the `hoogle` library, so that these modules only need to be built once?

Greetings,
Joachim

**Comment by ndmitchell** [comment]
Yep, that makes sense. Although I've rewritten lots of Hoogle in the repo, so don't intend to both making minor tweaks to Hoogle 4. I'll make sure I do that for Hoogle 5 though.

**Comment by ndmitchell** [comment]
Now done - the library exports a `hoogle :: [String] -> IO ()` function which I just call from the executable.
